test/index.js

```diff
@@ -23,51 +24,52 @@ const promisifyApi = (o) => {
   }
 }
 
+const when = () => {
+  var done = () => { throw Error('did not happen') }
+  const fn = () => done()
+  fn.done = promisify((cb) => { done = cb })
+  return fn
+}
+
 function validSocket (s) {
   if (!s) return false
-  return (s instanceof Socket) || (s._utp && s._utp instanceof UTP)
+  return (s instanceof net.Socket) || (s._utp && s._utp instanceof UTP)
 }
 
-test('socket recieved – socket option – (address details based connection)', async ({is, plan}) => {
-  // there is an unfixed stateful bug which makes
-  // this test non-atomic causing the socket method to never be called. 
-  // Temporary workaround is to place this test first – TODO: move rest to above 
-  // other socket recieved test
-  // NOTE: it may be utp-native, as even stripping all modules except utp-native 
-  // out of the require cache to get a fresh state fails. We can't strip utp-native
-  // from the require cache, it leads to an error (probably to do with Node expecations
-  // around native bindings, error is Module did not self register)
+test('network bound – bind option', async ({ pass, plan }) => {
   plan(1)
   const network = promisifyApi(guts({
-    socket(sock) {
-      is(validSocket(sock), true)
-    }
+    bind () { pass('method called') }
   }))
   await network.bind()
-  const client = promisifyApi(guts())
-  const { port } = network.address()
-  const host = '127.0.0.1'
-  await client.connect({ port, host })
   await network.close()
-  await client.close()
 })
 
-
-test('network bound – bind option', async ({pass, plan}) => {
+test('network closed – close option', async ({ pass, plan }) => {
   plan(1)
   const network = promisifyApi(guts({
-    bind() { pass('method called') }
+    close () { pass('method called') }
   }))
   await network.bind()
   await network.close()
 })
-test('network closed – close option', async ({pass, plan}) => {
-  plan(1)
-  const network = promisifyApi(guts({
-    close() { pass('method called') }
-  }))
-  await network.bind()
-  await network.close()
+
+test('announce before bind will throw', async ({ throws }) => {
+  const network = guts()
+  const topic = randomBytes(32)
+  throws(() => network.announce(topic), Error('Bind before announcing'))
```

Comment:
we should run our own dht bootstrapper for testing so we don't spam the global one

Reply:
it's straight forward. spin up a https://github.com/hyperswarm/dht instance and pass `bootstrap: [ '127.0.0.1:the-port' ]` to https://github.com/hyperswarm/guts/blob/master/index.js#L129 (we need to expose that func)

Reply:
Oh we are testing error cases here. Still would be good if all tests did this

Reply:
done